plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,474 @@
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ValueSource
+import org.gradle.api.provider.ValueSourceParameters
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters.None
+import org.gradle.api.tasks.Input
+import org.gradle.execution.RunRootBuildWorkBuildOperationType
+import org.gradle.internal.operations.BuildOperationDescriptor
+import org.gradle.internal.operations.BuildOperationListener
+import org.gradle.internal.operations.OperationFinishEvent
+import org.gradle.internal.operations.OperationIdentifier
+import org.gradle.internal.operations.OperationProgressEvent
+import org.gradle.internal.operations.OperationStartEvent
+import org.gradle.process.ExecOperations
+import org.gradle.util.GradleVersion
+
+abstract class SentryTelemetryService :
+    BuildService<None>,
+    BuildOperationListener,
+    AutoCloseable {
+
+    private var hub: IHub = NoOpHub.getInstance()
+    private var transaction: ITransaction? = null
+    private var didAddChildSpans: Boolean = false
+    private var started: Boolean = false
+
+    fun start(startParameters: SentryTelemetryServiceParams) {
+        if (started) {
+            return
+        }
+
+        started = true
+        try {
+            if (startParameters.saas == false) {
+                SentryPlugin.logger.info(
+                    "Sentry is running against a self hosted instance. Telemetry has been disabled."
+                )
+                hub = NoOpHub.getInstance()
+            } else if (!startParameters.sendTelemetry) {
+                SentryPlugin.logger.info("Sentry telemetry has been disabled.")
+                hub = NoOpHub.getInstance()
+            } else {
+                if (!Sentry.isEnabled()) {
+                    SentryPlugin.logger.info(
+                        "Sentry telemetry is enabled. To disable set `telemetry=false` " +
+                            "in the sentry config block."
+                    )
+                    Sentry.init { options ->
+                        options.dsn = startParameters.dsn
+                        options.isDebug = true
+                        options.isEnablePrettySerializationOutput = false
+                        options.tracesSampleRate = 1.0
+                        options.release = BuildConfig.Version
+                        options.isSendModules = false
+                        options.environment = startParameters.buildType
+                        options.setTag("SDK_VERSION", BuildConfig.SdkVersion)
+                        options.setTag("AGP_VERSION", AgpVersions.CURRENT.toString())
+                        options.setTag("BUILD_SYSTEM", "gradle")
+                        options.setTag("BUILD_TYPE", startParameters.buildType)
+                        options.setTag("GRADLE_VERSION", GradleVersion.current().version)
+                        startParameters.cliVersion?.let { options.setTag("SENTRY_CLI_VERSION", it) }
```

Comment:
```suggestion
                        options.setTag("SENTRY_CLI_VERSION", BuildConfig.CliVersion)
```
forgot to change it here actually, maybe we don't need it in `startParameters` then, if it's static

Reply:
Is there no way to override it?

Reply:
afaik you can set `cli.executable` in the properties file and it'll override what we have in `BuildConfig`, so I guess we should keep it as is?

Reply:
oh, I wasn't aware this is possible, but looks like it is https://github.com/getsentry/sentry-android-gradle-plugin/blob/dd4a7de205b4bfb2c035c0bf6af36fd4840a9164/plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryCliProvider.kt#L23-L26

Yeah this should be reverted then, sorry for that!